incubator/kafka/templates/configmap-config.yaml

```diff
@@ -0,0 +1,38 @@
+{{- if .Values.topics -}}
+{{- $zk := include "zookeeper.url" . -}}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  labels:
+    app: {{ template "kafka.fullname" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    heritage: "{{ .Release.Service }}"
+    release: "{{ .Release.Name }}"
+  name: {{ template "kafka.fullname" . }}-config
+data:
+  runtimeConfig.sh: |
+    #!/bin/sh
+    set -e
+    cd /usr/bin
+    until kafka-configs --zookeeper {{ $zk }} --entity-type topics --describe || (( count++ >= 6 ))
+    do
+      echo "Waiting for Zookeeper..."
+      sleep 20
+    done
+    echo "Applying runtime configuration using {{ .Values.image }}:{{ .Values.imageTag }}"
+    {{- range $n, $topic := .Values.topics }}
+    {{- if and $topic.partitions $topic.replicationFactor }}
+    kafka-topics --zookeeper {{ $zk }} --create --if-not-exists --force --topic {{ $topic.name }} --partitions {{ $topic.partitions }} --replication-factor {{ $topic.replicationFactor }}
+    {{- end }}
+    {{- if $topic.partitions }}
+    kafka-topics --zookeeper {{ $zk }} --alter --force --topic {{ $topic.name }} --replication-factor {{ $topic.replicationFactor }} || true
```

Comment:
it looks like this command is altering replication-factor -- should be partitions

Reply:
Good catch.  Although I'm now having second thoughts about including this.  To change number of partitions for keyed topics you need to understand the implications.  I guess I can put a warning in the docs.